test/unit/modules/blocks/process.js

```diff
@@ -0,0 +1,355 @@
+
+			blocksProcess.loadBlocksOffset(1, 10, true, function (err, loadedBlock) {
+				if (err) {
+					expect(err).equal('Failed to validate vote schema: Array items are not unique (indexes 0 and 4)');
+					return done();
+				}
+				
+				done(loadedBlock);
+			});
+		});
+
+	});
+
+	describe('loadBlocksOffset {verify: false} - rerun block/trs errors', function () {
+		it('should clear db for errors', function (done) {
+			async.every([
+				'forks_stat'
+			], function (table, seriesCb) {
+				clearDatabaseTable(db, modulesLoader.logger, table, seriesCb);
+			}, function (err, result) {
+				if (err) {
+					done(err);
+				}
+				done();
+			});
+		});
+
+		it('should load and process block 4 from db with blockSignature error', function (done) {
+			blocks.lastBlock.set(loadTables[0].data[1]);
+
+			blocksProcess.loadBlocksOffset(1, 4, false, function (err, loadedBlock) {
+				if (err) {
+					return done(err);
+				}
+				
+				expect(loadedBlock.id).equal(loadTables[0].data[2].id);
+				expect(loadedBlock.previousBlock).equal(loadTables[0].data[2].previousBlock);
+				done();
+			});
+		});
+
+		it('should load and process block 5 from db with invalid payloadHash', function (done) {
+			blocks.lastBlock.set(loadTables[0].data[2]);
+
+			blocksProcess.loadBlocksOffset(1, 5, false, function (err, loadedBlock) {
+				if (err) {
+					return done(err);
+				}
+
+				expect(loadedBlock.id).equal(loadTables[0].data[3].id);
+				expect(loadedBlock.previousBlock).equal(loadTables[0].data[3].previousBlock);
+				done();
+			});
+		});
+
+		it('should load and process block 6 from db with invalid block timestamp', function (done) {
+			blocks.lastBlock.set(loadTables[0].data[3]);
+
+			blocksProcess.loadBlocksOffset(1, 6, false, function (err, loadedBlock) {
+				if (err) {
+					done(err);
+				}
+
+				expect(loadedBlock.id).equal(loadTables[0].data[4].id);
+				expect(loadedBlock.previousBlock).equal(loadTables[0].data[4].previousBlock);
+				done();
+			});
+		});
+
+		it('should load block 7 from db and return error unknown transaction type', function (done) {
+			blocks.lastBlock.set(loadTables[0].data[4]);
+
+			blocksProcess.loadBlocksOffset(1, 7, true, function (err, loadedBlock) {
+				if (err) {
+					expect(err).equal('Blocks#loadBlocksOffset error: Unknown transaction type 99');
+					return done();
+				}
+				
+				done(loadedBlock);
+			});
+		});
+
+		it('should load and process block 8 from db with Invalid block version error', function (done) {
+			blocks.lastBlock.set(loadTables[0].data[5]);
+
+			blocksProcess.loadBlocksOffset(1, 8, false, function (err, loadedBlock) {
+				if (err) {
+					done(err);
+				}
+				
+				expect(loadedBlock.id).equal(loadTables[0].data[6].id);
+				expect(loadedBlock.previousBlock).equal(loadTables[0].data[6].previousBlock);
+				done();
+			});
+		});
+
+		it('should load and process block 9 from db with Invalid previous block error (no fork:1)', function (done) {
```

Comment:
`should load and process block 9 from db with invalid previousBlock (no fork:1)`